packages/wrangler/src/pages/deploy.tsx

```diff
@@ -337,9 +345,65 @@ export const Handler = async (args: PagesDeployArgs) => {
 		project_name: projectName,
 	});
 
-	logger.log(
-		`✨ Deployment complete! Take a peek over at ${deploymentResponse.url}`
-	);
+	let latestDeploymentStage: DeploymentStage | undefined;
+	let attempts = 0;
+
+	logger.log("🌎 Checking deployment status...");
+
+	while (
+		attempts < MAX_DEPLOYMENT_STATUS_ATTEMPTS &&
+		latestDeploymentStage?.name !== "deploy" &&
+		latestDeploymentStage?.status !== "success" &&
+		latestDeploymentStage?.status !== "failure"
+	) {
+		try {
+			/*
+			 * Exponential backoff
+			 * On every retry, exponentially increase the wait time: 1 second, then
+			 * 2s, then 4s, then 8s, etc.
+			 */
+			await new Promise((resolvePromise) =>
+				setTimeout(resolvePromise, Math.pow(2, attempts++) * 1000)
+			);
+
+			logger.debug(
+				`attempt #${attempts}: Attempting to fetch status for deployment with id "${deploymentResponse.id}" ...`
+			);
+
+			const deployment = await fetchResult<Deployment>(
+				`/accounts/${accountId}/pages/projects/${projectName}/deployments/${deploymentResponse.id}`
+			);
+			latestDeploymentStage = deployment.latest_stage;
+		} catch (err) {
+			// don't retry if API call retruned an error
+			logger.debug(
+				`Attempt to get deployment status for deployment with id "${deploymentResponse.id}" failed: ${err}`
+			);
+		}
+	}
+
+	if (latestDeploymentStage?.status === "success") {
```

Comment:
We probably want to double check that `name === "deploy"` here rather than "build"

Reply:
that's a very good callout!

Reply:
✅